From 7b87320aab69289cfab4493e0efe53ad3ee90c78 Mon Sep 17 00:00:00 2001
From: Chen Meng J <meng.j.chen@intel.com>
Date: Tue, 11 Sep 2018 13:52:50 +0800
Subject: [PATCH 2263/2302] media: intel-ipu4: ti960: fix build warning

Message for Open Source:
fix build warning.

Message for Internal:
remove unused function ti953_reg_read.

Change-Id: Id8e9a0dd4b1e51ca58f35432d94b4a089f06a28a
Signed-off-by: Chen Meng J <meng.j.chen@intel.com>
Reviewed-on: ICG #354617
Reviewed-by: Coding Style
Approver: Wei Meng <wei.meng@intel.com>
Reviewed-by: Ying Chang <ying.chang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/ti960.c | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/drivers/media/i2c/ti960.c b/drivers/media/i2c/ti960.c
index 59d2593..198f37f 100644
--- a/drivers/media/i2c/ti960.c
+++ b/drivers/media/i2c/ti960.c
@@ -149,33 +149,6 @@ static int ti953_reg_write(struct ti960 *va, unsigned short rx_port,
 	return 0;
 }
 
-static int ti953_reg_read(struct ti960 *va, unsigned short rx_port,
-	unsigned char reg, unsigned char *val)
-{
-	int retry, timeout = 10;
-	struct i2c_client *client = v4l2_get_subdevdata(&va->sd);
-	unsigned short ser_alias = va->pdata->subdev_info[rx_port].ser_alias;
-
-	client->addr = ser_alias;
-	for (retry = 0; retry < timeout; retry++) {
-		*val = i2c_smbus_read_byte_data(client, reg);
-		if (*val < 0)
-			usleep_range(5000, 6000);
-		else
-			break;
-	}
-
-	client->addr = TI960_I2C_ADDRESS;
-	if (retry >= timeout) {
-		dev_err(va->sd.dev,
-			"%s:read reg failed: port=%2x, addr=%2x, reg=%2x\n",
-			__func__, rx_port, ser_alias, reg);
-		return -EREMOTEIO;
-	}
-
-	return 0;
-}
-
 static int ti960_reg_read(struct ti960 *va, unsigned char reg, unsigned int *val)
 {
 	int ret, retry, timeout = 10;
-- 
2.7.4

